backend/src/main/repositories/CallRepository.ts

```diff
@@ -0,0 +1,134 @@
+            createdCall.save();
+
+            log(`Added new call id ${createdCall.id}`);
+            return Promise.resolve(createdCall);
+        } catch (err: any) {
+            log(err);
+            return Promise.reject(err);
+        }
+    };
+
+    public delete = async (id: number): Promise<number> => {
+        try {
+            const deletedCallStatus = await Call.destroy({
+                where: {
+                    id: id
+                }
+            });
+
+            log(`Call with id ${id} has been deleted`);
+            return Promise.resolve(deletedCallStatus);
+        } catch (err: any) {
+            log(err);
+            return Promise.resolve(err);
+        }
+    };
+
+    public update = async (id: number, updatedValue: CallUpdateDTO): Promise<number> => {
+        try {
+            await Call.update(updatedValue, {
+                where: {
+                    id: id
+                }
+            });
+
+            log(`Call with id ${id} has been updated`);
+            return Promise.resolve(1);
+        } catch (err: any) {
+            return Promise.reject(err);
+        }
+    };
+
+    public get = async (specificCall: { id: number, email: string }): Promise<Call | null> => {
+        try {
+            const call = await Call.findOne({
+                where: {
+                    id: specificCall.id,
+                    email: specificCall.email
+                }
+            });
+
+            log(`Call with id ${call?.id} and email ${call?.email} has been retrieved`);
+            return Promise.resolve(call);
+        } catch (err: any) {
+            log(err);
+            return Promise.reject(err);
+        }
+    };
+
+    public getByEmail = async (email: string): Promise<Call | null> => {
+        try {
+            const call = await Call.findOne({
+                where: {
+                    email: email
+                }
+            });
+
+            log(`Call with id ${call?.id} and email ${call?.email} has been retrieved`);
+            return Promise.resolve(call);
+        } catch (err: any) {
+            log(err);
+            return Promise.reject(err);
+        }
+    };
+
+    public getById = async (id: number): Promise<Call | null> => {
+        try {
+            const call = await Call.findOne({
+                where: {
+                    id: id
+                }
+            });
+
+            log(`Call with id ${call?.id} and email ${call?.email} has been retrieved`);
+            return Promise.resolve(call);
+        } catch (err: any) {
+            log(err);
+            return Promise.reject(err);
+        }
+    };
+
+    public getAllByEmail = async (email: string): Promise<Call[] | null> => {
+        try {
+            const call: Call[] = await Call.findAll({
+                where: {
+                    email: email
+                }
+            });
+
+            log(`${call} has been retrieved`);
```

Comment:
Not sure if logging an array would be pretty ? Correct me if I'm wrong.

Reply:
Reworded the log message